blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java

```diff
@@ -113,6 +113,10 @@ private static void validate(AggregatedPayload payload) {
                 metricName = ((BluefloodSet) leafBean).getName();
             }
 
+            if ((metricName.isEmpty()) && (payload.getAllMetricNames().size() > 0)) {
+                metricName = payload.getAllMetricNames().get(0);
+            }
```

Comment:
does the first metricName in the list always relate to the current violation?  If there isn't a metricName, does that mean that it couldn't be processed, and should be empty?